### DISCUSSION TITLE
how to implement hovercard with caching in livewire ?

### DISCUSSION THREAD

**Discussion by ariaieboy** [livewire/livewire#4403] [1. Help]
hello everyone 
in github when you hover on somebody's avatar github will send request to the server and this request will return the user info.
github named this card to hovercard. 
the interesting part of this feature is that when you hover a user avatar for first time in a page github will send the request but if you move your mouse and hover the avatar again github will show you the hover card without sending another request to get the hovercard rendered info from server. 
now i want know that can we implement such thing in livewire or not ? 

the important part is the single request for each hovercard.

and its not only the hovercard that github use this approach.
another example is the new custom star list that github added. when you visiting a project if you click on the star button a request will send to the server that retrive the custom star lists dropdown and if you close the dropdown and re-click on it the dropdown will pop up very fast without any http request and loading time.

**Comment by ariaieboy** [comment]
@joshhanley @calebporzio 
please read this discussion.
I think if livewire support this ,you guys should add some document about it.
and if livewire does not support this way of loading and caching the components we must work on this feature. 

because with using of this technique we can send a very small rendered html to user and then depends on user action render and show the component we want without sending millions of extra requests to server. 

and another thing that is needed for livewire is a performance best practise document. that teach livewire users how to build performant application using livewire

**Comment by joshhanley** [comment]
@ariaieboy Livewire already has a similar ability with prefetching https://laravel-livewire.com/docs/2.x/prefetching

Which should do what you're asking. Can you test that and let us know how it goes?

Thanks!

**Reply by ariaieboy** [reply]
yeah i tested the prefetching but its not what i am looking for.
i think what we want is wire:append that is in the road map of the livewire.

what i want is to define a component that is not loaded in inital request. but it will load and append to the page after user trigger an event

**Comment by joshhanley** [comment]
@ariaieboy ah right yeah that makes sense.

I wonder whether you could do that just behind a flag using Alpine, so have an entangled flag like `showCard` or something, then on hover, if Alpine's `showCard` prop is false, then set it to true, which would trigger a LW request which would then show the card on the page? But on next hover the card html would already be loaded so it would just display the card.

Hope this helps!

**Reply by ariaieboy** [reply]
yes with alpine we can implement this feature. 

like u said we can add a flag and in the first attemp we can get the rendered html from LW using this `let baz = await $wire.bar(..args)` from a defined method

but i just looking for sth more internally than using alpine and hack the lifecycle and how LW works.

since livewire is here to work as a fullstack framework i think we should cover this edge cases.

specially in V3.

**Reply by joshhanley** [reply]
@ariaieboy although Livewire is fullstack, I think for interactive stuff like this, the best use is to use Alpine with Livewire to get the effect you want 🙂

V3 will actually change Livewire's frontend to become an Alpine plugin, so to me using Alpine is a good solution for this. Ideally once that happens though, someone could make an Alpine/Livewire plugin that supports this, to make it easy to use.
